### ISSUE TITLE
Way to put a bounty on a specific issue/milestone/project

### ISSUE THREAD

**Issue by XVilka** [opencollective/opencollective#2420]
## User story 
If some sponsor of organization core contributor wants to sponsor a particular problem, or core contributor wants to put a bounty as an expense to a particular problem OpenCollective platform should allow to do that, something similar to BountySource.

## Best solution for this problem
At first, there should be a way to select GitHub or GitLab targets for both sponsors and organization contributors. Possible targets are:
- An individual issue
- Milestone
- Project (from repository or GitHub/GitLab organization)

**Comment by piamancini** [comment]
yes! we actually had a collective (dokku) and a host (oscollective.org) funding issues. We want to add this to our bounty system in the future. cc @xdamman @znarf

**Comment by gusaus** [comment]
CiviCRM has a few sustainability initiatives that might serve as inspiration for this type of functionality. 

* [Make It Happen (MIH)](https://civicrm.org/make-it-happen) - campaigns to get or support new features that you’d like to see into the project but which hasn't been included within core development
* [Paid Issue Queue](https://civicrm.org/paid-issue-queue) - letting people with the money jump the queue and get the fixes they needed into project
* [Support for bug fixes](https://civicrm.org/fix) - funding priority bug fixes

**Comment by gusaus** [comment]
I think a good start might be to work some processes for financially contributing to bounties into the documentation - 
https://docs.opencollective.com/help/contributing/development/bounties

Maybe Open Collective could assign a bounty for that documentation task!

BountySource also provides some good inspiration in terms of process - 
https://github.com/bountysource/core/wiki/Frequently-Asked-Questions

**Comment by XVilka** [comment]
You can see also how it is implemented in this simple FOSS bounty system, that supports multiple cryptocurrencies and work with GitHub issues - [jollheef/donate](https://github.com/jollheef/donate).

This is how it looks [in the GitHub issue](https://github.com/jollheef/donate/issues/8)

![image](https://user-images.githubusercontent.com/203261/73710690-a978cf00-473f-11ea-80be-a2e62dad2f27.png)

**Comment by alanna** [comment]
I think this could be achieved right now through using standalone tier pages with and issue as the goal. Pick an issue to raise funds for and people can contribute to it specifically.

**Comment by gusaus** [comment]
> I think this could be achieved right now through using standalone tier pages with and issue as the goal. Pick an issue to raise funds for and people can contribute to it specifically.

Do we know what collective would be used to create tiers? It seems totally doable if we can define a process.  https://github.com/opencollective/opencollective/issues/2420#issuecomment-557367715

Possibly we can incorporate some of the following?   https://github.com/opencollective/opencollective/issues/2503 https://github.com/opencollective/opencollective/issues/2420#issuecomment-531527636

**Comment by RCheesley** [comment]
Would also be interested in this - we are currently using BountySource but would love to keep it all central in Open Collective.

**Comment by poikilos** [comment]
> You can see also how it is implemented in this simple FOSS bounty system, that supports multiple cryptocurrencies and work with GitHub issues - [jollheef/donate](https://github.com/jollheef/donate).
> 
> This is how it looks [in the GitHub issue](https://github.com/jollheef/donate/issues/8)
> 
> ![image](https://user-images.githubusercontent.com/203261/73710690-a978cf00-473f-11ea-80be-a2e62dad2f27.png)

@XVilka The links are broken. Does anyone know where this went or whether it still works or is maintained anywhere?

**Comment by XVilka** [comment]
@poikilos no, it seems not maintained anymore.

**Comment by pktiuk** [comment]
This is a very useful feature for small teams developing apps (or single developers developing apps on their own).    
These kinds of bug bounties may help to attract contributors from outside.
